### ISSUE TITLE
🙋 Allow to specify a hostname for hosted IDEs

### ISSUE THREAD

**Issue by jessehattabaugh** [parcel-bundler/parcel#276]
I use [Cloud9](c9.io) which is a hosted IDE site. They do some host/port mapping stuff so they want me to use an environment variable whenever I start a server. I understand that I can pass the `$PORT` variable to the `-p` argument, but there doesn't appear to be an argument for the host, or ip address which I could pass the `$IP` variable to. 

### 💁 Possible Solution
`npx parcel-bundler index.html -p $PORT -h $IP`

**Comment by brandon93s** [comment]
@jessehattabaugh - Please see #278 for further discussion. The default host aligns with the host that Cloud9 specifies in their [docs](https://docs.c9.io/docs/run-an-application), so parcel should "just work" in Cloud9 without specifying a host. If this is not the case, please reply and we'll work through a solution.

**Comment by Fluidbyte** [comment]
@brandon93s I'm having a similar issue, perhaps I'm missing something here. I work on a remote server (i.e. not `localhost`) and starting the parcel HMR works, the trouble is the socket connection is looking for `http://localhost:<ws-port>` which means the only way I've been able to get it working is to point `localhost` in my local hosts file to my server.

**Comment by brandon93s** [comment]
I see. I had approached this from the perspective of the HTTP server and not the implications on the HMR socket. Will re-open for further investigation.

**Comment by leonardosnt** [comment]
I'm having the same issue here. 
I'm using a smartphone to connect to the website over LAN and the HMR fails to establish the connection
because the WebSocket is connecting to the localhost instead of `window.location.hostname`.

I fixed this issue by forking parcel and changing this:
https://github.com/parcel-bundler/parcel/blob/0eb4487491fd70390697ad413aeac994fca4309c/src/builtins/hmr-runtime.js#L18
to:
```js
var ws = new WebSocket('ws://' + window.location.hostname + ':{{HMR_PORT}}/');
```

**Comment by Fluidbyte** [comment]
This seemed like a pretty quick fix for an issue that can cause a lot of headaches, opened #426 above thanks to @leonardosnt's comment.

**Comment by scriptPilot** [comment]
Unfortunately, I still have an issue in Cloud9 at Amazon Web Services.

* File src/client/index.html links to file src/client/index.js
* File index.js should log some text to the console
* Command to start server is `parcel src/client/index.html -p 8080`

Now, it's not possible to open the server by the URL given by Parcel console log - but with the one given by Cloud9 for view in a separate window.

But the following error appears:

![bildschirmfoto 2018-01-04 um 22 09 06](https://user-images.githubusercontent.com/19615586/34584428-05226d6c-f19c-11e7-8f08-383a74fd7533.png)

**Comment by Fluidbyte** [comment]
I don't think this actually made it out in a release yet, unless you're specifically using `master`. Additionally, you'll need to specify both the port `-p` and hmr-port `-h`

**Comment by scriptPilot** [comment]
Last release was 4 days ago, the code commits about 5 days ... mhm, maybe I am wrong, but I thought the code is already published.

**Comment by Fluidbyte** [comment]
It's not in the latest (1.4.1), so you'll need to pull from `master`

**Comment by scriptPilot** [comment]
Mhm, wenn I install from parcel-bundler/parcel (master), I get the following error:

```
> node_modules/.bin/parcel src/index.html

module.js:471
    throw err;
    ^

Error: Cannot find module '../lib/cli'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/ubuntu/workspace/test-parcel/node_modules/parcel-bundler/bin/cli.js:6:7)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
```

**Comment by devongovett** [comment]
Closing. Should be fixed by #426.

**Comment by thinkloop** [comment]
Hello, 

I am not able to make parcel work in cloud 9 out of the box. I am running:

```
parcel ./src/index.html --https --port $PORT --hmr-hostname $IP --no-hmr -d ./parcel
```

The error message from cloud 9:

`
You may be using the wrong PORT & IP for your server application.Try passing $PORT and $IP to properly launch your application.
`

I was looking for some sort of '--host' parameter which I could pass $IP to. I am not looking for hmr, just to run the basic site. I'm running `"parcel-bundler": "1.6.2"`.
